dataikuapi/dss/utils.py

```diff
@@ -257,4 +260,9 @@ def custom_fields(self):
 
     @custom_fields.setter
     def custom_fields(self, custom_fields):
-        self._tod["customFields"] = custom_fields
\ No newline at end of file
+        self._tod["customFields"] = custom_fields
+
+
+def extract_info_from_full_model_id(full_model_id):
+    match = re.search(u'([\w]+)-([\w]+)-([\w]+)-s\w+-pp\w+-m\w+', full_model_id)
+    return match.group(1), match.group(2), match.group(3)
```

Comment:
- Only used once, can inline for now. A proper generic function would need to be more though out, potentially use capturing group names and return the full structure, or even a custom object equivalent to backend `FullModelId`, in which case this would be a static factory method…
- In context, this can only return analysis trained models, so it should
    - match the whole string
    - enforce the initial `A-`

Reply:
I'm not sure I understand all, but this method is used twice.

Reply:
Ah true, but in both cases we treat it exactly the same. In this case, you can move this as `MLTask.from_full_model_id(fmi)`. (And second point still stands)

Reply:
If we want to override project key, can add that as an optional argument